### PR TITLE
Fix is_isomorphous

### PR DESCRIPTION
**Pull request by kmdalton**
As discussed in #277 , DataSet.is_isomorphous currently does not do what is indicated in its docstring. This change makes the behavior in line with the expectation that `cell_threshold` is a percent difference.

### CODE REVIEW

**Review by tjlane — APPROVED** [review]
Bug is fixed.

I'd like to see a (regression) test for this method added before merging, to ensure it *stays* fixed.

**Comment by kmdalton** [comment]
as requested, @tjlane

File: tests/test_dataset.py [hunk 1]
```diff
@@ -603,6 +603,40 @@ def test_is_isomorphous(data_unmerged, data_fmodel, sg1, sg2, cell1, cell2):
...
+        if isomorphous:  # shrink
+            factor = factor * (1.0 - epsilon)
+        else:  # grow
+            factor = factor * (1.0 + epsilon)
+    else:  # cell_test < cell
+        factor = (200.0 - threshold) / (200.0 + threshold)
+        if isomorphous:  # grow
+            factor = factor * (1.0 + epsilon)
+        else:  # shrink
+            factor = factor * (1.0 - epsilon)
```
Comment (tjlane) [inline tests/test_dataset.py:631]: This test is smart... maybe too smart? It had to read it quite a few times before seeing it clearly.

Generally I like dumb stuff. More like:

```python
if isomorphous:
    cell_change_factor = (100.0 + threshold) / 100.0 - epsilon
else:
    cell_change_factor = (100.0 + threshold) / 100.0 + epsilon

if sign:
    cell_test = cell * cell_change_factor
else:
    cell_test = cell / cell_change_factor
```

or something similar.

File: tests/test_dataset.py [hunk 2]
```diff
@@ -603,6 +603,40 @@ def test_is_isomorphous(data_unmerged, data_fmodel, sg1, sg2, cell1, cell2):
             assert not result
 
 
+@pytest.mark.parametrize("threshold", [5.0, 1.0, 0.5, 0.1])
+@pytest.mark.parametrize("isomorphous", [True, False])
+@pytest.mark.parametrize("sign", [True, False])  # Which cell is bigger
```
Comment (tjlane) [inline tests/test_dataset.py:608]: `sign` is a bit confusing, I anticipated something being positive or negative.

What about `@pytest.mark.parametrize("compare_to", ["bigger_cell", "smaller_cell"])`, then compare strings in what follows?

**Review by tjlane — COMMENTED** [review]
Nice! Logic looks solid. A few comments in the interest of people who might read this code in the future.

Also: do we need a test for the spacegroup comparison bit?

File: tests/test_dataset.py (hunk 2 above)
Comment (kmdalton) [reply tests/test_dataset.py:608]: Okay I clarified it.

File: tests/test_dataset.py (hunk 1 above)
Comment (kmdalton) [reply tests/test_dataset.py]: Okay I think I made it dumber. Let me know.

**Comment by kmdalton** [comment]
@tjlane , space groups are tested in  `test_is_isomorphous`. I don't think there is any point in doing it for this test too.

**Review by tjlane — APPROVED** [review]
love it

**Comment by tjlane** [comment]
thanks for tackling this one @kmdalton !

**Pull request merged**